src/test/mir-opt/if_condition_int.test2.SimplifyComparisonIntegral.diff

```diff
@@ -0,0 +1,91 @@
+      let mut _0: u32;                     // return place in scope 0 at $DIR/if-condition-int.rs:39:21: 39:24
+      let mut _2: bool;                    // in scope 0 at $DIR/if-condition-int.rs:40:8: 40:15
+      let mut _3: u32;                     // in scope 0 at $DIR/if-condition-int.rs:40:8: 40:9
+      let mut _4: bool;                    // in scope 0 at $DIR/if-condition-int.rs:42:15: 42:22
+      let mut _5: u32;                     // in scope 0 at $DIR/if-condition-int.rs:42:15: 42:16
+  
+      bb0: {
+          StorageLive(_2);                 // scope 0 at $DIR/if-condition-int.rs:40:8: 40:15
+          StorageLive(_3);                 // scope 0 at $DIR/if-condition-int.rs:40:8: 40:9
+          _3 = _1;                         // scope 0 at $DIR/if-condition-int.rs:40:8: 40:9
+-         _2 = Eq(move _3, const 42_u32);  // scope 0 at $DIR/if-condition-int.rs:40:8: 40:15
+-                                          // ty::Const
+-                                          // + ty: u32
+-                                          // + val: Value(Scalar(0x0000002a))
+-                                          // mir::Constant
+-                                          // + span: $DIR/if-condition-int.rs:40:13: 40:15
+-                                          // + literal: Const { ty: u32, val: Value(Scalar(0x0000002a)) }
+-         StorageDead(_3);                 // scope 0 at $DIR/if-condition-int.rs:40:14: 40:15
+-         switchInt(_2) -> [false: bb1, otherwise: bb2]; // scope 0 at $DIR/if-condition-int.rs:40:5: 46:6
++         switchInt(move _3) -> [42_u32: bb2, otherwise: bb1]; // scope 0 at $DIR/if-condition-int.rs:40:5: 46:6
+      }
+  
+      bb1: {
++         StorageDead(_3);                 // scope 0 at $DIR/if-condition-int.rs:40:5: 46:6
+          StorageLive(_4);                 // scope 0 at $DIR/if-condition-int.rs:42:15: 42:22
+          StorageLive(_5);                 // scope 0 at $DIR/if-condition-int.rs:42:15: 42:16
+          _5 = _1;                         // scope 0 at $DIR/if-condition-int.rs:42:15: 42:16
+-         _4 = Ne(move _5, const 21_u32);  // scope 0 at $DIR/if-condition-int.rs:42:15: 42:22
+-                                          // ty::Const
+-                                          // + ty: u32
+-                                          // + val: Value(Scalar(0x00000015))
+-                                          // mir::Constant
+-                                          // + span: $DIR/if-condition-int.rs:42:20: 42:22
+-                                          // + literal: Const { ty: u32, val: Value(Scalar(0x00000015)) }
+-         StorageDead(_5);                 // scope 0 at $DIR/if-condition-int.rs:42:21: 42:22
+-         switchInt(_4) -> [false: bb3, otherwise: bb4]; // scope 0 at $DIR/if-condition-int.rs:42:12: 46:6
++         switchInt(move _5) -> [21_u32: bb3, otherwise: bb4]; // scope 0 at $DIR/if-condition-int.rs:42:12: 46:6
```

Comment:
A further optimization would be to fold the targets of this switch into the terminator of bb0. That could be a separate PR though?

Reply:
yes, please leave it to a separate PR